json-rpc/src/stream_rpc/counters.rs

```diff
@@ -0,0 +1,148 @@
+        &[
+            "transport", // transport used: ws, ipc, etc
+            "method", // method of request, matches JSON RPC method name (e.g. "submit", "get_account")
+            "result", // result of request: "success", "fail"
+            "sdk_lang", // the language of the SDK: "java", "python", etc
+            "sdk_ver", // the version of the SDK: "0.0.11", "1.45.31", etc
+        ]
+    )
+        .unwrap()
+});
+
+/// number of messages sent. These are in response to subscriptions.
+pub static SUBSCRIPTION_RPC_SENT: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_streaming_rpc_subscription_rpc_sent",
+        "Cumulative number of RPC messages that the Streaming RPC service sends as part of subscriptions",
+        &[
+            "transport", // transport used: ws, ipc, etc
+            "method", // method of request, matches JSON RPC method name (e.g. "submit", "get_account")
+            "result", // result of request: "success", "fail"
+            "sdk_lang", // the language of the SDK: "java", "python", etc
+            "sdk_ver", // the version of the SDK: "0.0.11", "1.45.31", etc
+        ]
+    )
+    .unwrap()
+});
+
+/// number of 'OK' messages sent. These are sent when a subscription is started.
+pub static SUBSCRIPTION_OKS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_streaming_rpc_subscription_oks",
+        "Cumulative number of 'OK' messages sent in response to starting a subscription",
+        &[
+            "transport", // transport used: ws, ipc, etc
+            "method", // method of request, matches JSON RPC method name (e.g. "submit", "get_account")
+            "result", // result of request: "success", "fail"
+            "sdk_lang", // the language of the SDK: "java", "python", etc
+            "sdk_ver", // the version of the SDK: "0.0.11", "1.45.31", etc
+        ]
+    )
+    .unwrap()
+});
+
+/// Cumulative number of clients connected
+pub static CLIENT_CONNECTED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_streaming_rpc_clients_count",
+        "Cumulative number of connected clients",
+        &[
+            "transport", // transport used: ws, ipc, etc
+            "sdk_lang",  // the language of the SDK: "java", "python", etc
+            "sdk_ver",   // the version of the SDK: "0.0.11", "1.45.31", etc
+        ]
+    )
+    .unwrap()
+});
+
+pub static INVALID_REQUESTS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_streaming_rpc_invalid_requests_count",
+        "Cumulative number of invalid requests that JSON RPC client service receives",
+        &[
+            "transport", // transport used: ws, ipc, etc
+            "method", // method of request, matches JSON RPC method name (e.g. "submit", "get_account")
+            "errortype", // categories of invalid requests: "invalid_format", "invalid_params", "invalid_method", "method_not_found"
```

Comment:
for some stuff like this, it might be beneficial to have a set of functions that provide some sort of type checking for logs and maybe even use enums so you get static checks.